common/strings.go

```diff
@@ -0,0 +1,21 @@
+package common
+
+import "strconv"
+
+// NewStringIfNotEmpty takes strings a and b, and returns a unless
+// string b is not empty.
+func NewStringIfNotEmpty(original string, newString string) string {
```

Comment:
This logic doesn't match the original intention.  Previously, things were only added to the cfn param map iff the string that was being set was not empty.  If it was empty previously, the key wouldn't even be set in the map, where now it is being added to the map with an empty string. 

I'm surprised no unit tests covered this so i looked and sure enough, the unit tests in the workflow package don't do any validation of the stack params that are being passed to the CFN manager mock.  Fortunately, `make e2e` did catch this.

Reply:
"thanks I hate it" I'll come up with an alternative